### PR TITLE
Upgrade libraries

### LINKED ISSUE #77: Upgrade libraries for JDK8

### PR DESCRIPTION
**Pull request by natan-cox-cognizone**
upgrade of libraries; 
replace junit tests with jupiter
fix error handling of index deletion; 
delete SimpleElasticRepository.java

fix #77

### CODE REVIEW

**Review by ssamarid — APPROVED** [review]

**Review by alexey-lukashov — APPROVED** [review]
Are you going to release A2 before merging this PR?

File: service/src/main/java/zone/cogni/asquare/service/index/TypeIndexService.java [hunk 1]
```diff
@@ -86,8 +83,10 @@ public void resetIndex(String indexName, ObjectNode indexSettings) {
       indexConfigProvider.getElasticStore().deleteIndex(indexName);
       log.info(".. index '{}' deleted", indexName);
     }
-    catch(IndexNotFoundException ex) {
-      log.info(".. index '{}' not found", indexName);
```
Comment (domivds) [inline service/src/main/java/zone/cogni/asquare/service/index/TypeIndexService.java:90]: I think we have to leave this here. On initial start the index does not exist, so this call should not fail if the index does not exist

File: build.gradle [hunk 2]
```diff
@@ -3,11 +3,14 @@ import pl.allegro.tech.build.axion.release.domain.scm.ScmPosition
...
 buildscript {
   ext {
-    guavaVersion = '23.0'
-    httpComponentsVersion = '4.5.12'
-    jenaVersion = '3.10.0'
-    springBootVersion = '2.2.9.RELEASE'
-    junitJupiterVersion = '5.5.2'
+    apachePoiVersion = '3.17'
+    commonsCodecVersion = '1.15'
+    commonsIoVersion = '2.7'
```
Comment (domivds) [inline build.gradle:8]: btw, doesn't spring by default includes a version of commons-io (and others), maybe we can just take these ones if that is the case

File: build.gradle [hunk 3]
```diff
@@ -80,10 +83,10 @@ subprojects {
...
-      dependency('org.apache.poi:poi:3.16')
-      dependency('org.apache.poi:poi-ooxml:3.16')
-      dependency('org.elasticsearch:elasticsearch:5.6.3')
-      dependency('org.elasticsearch.client:transport:5.6.3')
+      dependency("org.apache.poi:poi:$apachePoiVersion")
+      dependency("org.apache.poi:poi-ooxml:$apachePoiVersion")
+      dependency("org.elasticsearch:elasticsearch:$elasticsearchVersion")
+      dependency("org.elasticsearch.client:transport:$elasticsearchVersion")
       dependency('org.apache.commons:commons-pool2:2.11.1')
       dependency('cglib:cglib:3.3.0')
```
Comment (domivds) [inline build.gradle:91]: should we not put all versions in variables?

File: cube/src/main/java/zone/cogni/asquare/cube/index/AliasedCollectionIndexingService.java [hunk 4]
```diff
@@ -186,8 +185,10 @@ private void deleteIndex(@Nonnull IndexSwapState indexSwapState) {
     try {
       elasticStore.deleteIndex(oldIndexName);
     }
-    catch (IndexNotFoundException ignore) {
-      log.warn("(deleteIndex) index '{}' did not exist.", oldIndexName);
```
Comment (domivds) [inline cube/src/main/java/zone/cogni/asquare/cube/index/AliasedCollectionIndexingService.java:190]: same as above, should delete of non-existing index throw an exception?

File: cube/src/main/java/zone/cogni/asquare/cube/index/AliasedIndexIndexingService.java [hunk 5]
```diff
@@ -130,8 +129,10 @@ private void deleteIndex(@Nonnull IndexSwapState indexSwapState) {
     try {
       elasticStore.deleteIndex(oldIndexName);
     }
-    catch (IndexNotFoundException ignore) {
-      log.warn("(deleteIndex) index '{}' did not exist.", oldIndexName);
```
Comment (domivds) [inline cube/src/main/java/zone/cogni/asquare/cube/index/AliasedIndexIndexingService.java:134]: see above

File: cube/src/main/java/zone/cogni/asquare/cube/index/StaticIndexIndexingService.java [hunk 6]
```diff
@@ -150,8 +149,10 @@ private void deleteIndex(@Nonnull IndexFolder indexFolder) {
     try {
       elasticStore.deleteIndex(indexFolder.getName());
     }
-    catch (IndexNotFoundException ignore) {
-      log.info("(deleteIndex) index '{}' did not exist.", indexFolder.getName());
```
Comment (domivds) [inline cube/src/main/java/zone/cogni/asquare/cube/index/StaticIndexIndexingService.java:154]: see above

File: service/src/main/java/zone/cogni/asquare/service/elasticsearch/v7/Elasticsearch7AccessService.java [hunk 7]
```diff
@@ -291,8 +290,10 @@ public void resetIndex(ObjectNode indexSettings) {
       elasticStore.deleteIndex(indexName);
       log.info(".. index '{}' deleted", indexName);
     }
-    catch(IndexNotFoundException ex) {
-      log.info(".. index '{}' not found", indexName);
```
Comment (domivds) [inline service/src/main/java/zone/cogni/asquare/service/elasticsearch/v7/Elasticsearch7AccessService.java:295]: see above

File: build.gradle (hunk 2 above)
Comment (natan-cox-cognizone) [reply build.gradle:8]: yeah. I looked. seems spring does not like apache.

**Review by domivds — APPROVED** [review]

**Pull request merged**